### ISSUE TITLE
Cannot get example to work

### ISSUE THREAD

**Issue by neverfox** [RisingStack/graffiti-mongoose#79]
No matter what I try as a query, I get the following error in devtools (react.min.js):

```
Uncaught TypeError: Cannot read property 'getQueryType' of undefined
```

**Comment by dsifford** [comment]
@neverfox Can you provide an example of some of the queries you are trying?

**Comment by neverfox** [comment]
Well, I'm a total newbie when it comes to graphQL, but I was trying things from the examples like:

```
query UsersQuery {
  users {
    name
}
```

I assume that if was just badly formed, I'd get an error inside of graphiql, but it's just failing at the source code level somehow. Do you know of a query that you know should work that I could try?

**Comment by dsifford** [comment]
Hm.. I see what you mean.. 

The GraphiQL instance is not getting access to the database on my end. Tried to spin up the example on my end and no dice either. 

@neverfox are you able to open up the "Doc" sidebar and see the schema? If not, you probably have the same issue.

I'd usually troubleshoot, but I'm strapped for time and I'm not familiar with koa.

**Comment by neverfox** [comment]
Nope, no schema.

**Comment by tothandras** [comment]
Make sure that MongoDB is running.

**Comment by dsifford** [comment]
@tothandras Not sure that's the issue. MongoDB runs as a service on my machine (Ubuntu Wily) and I have no issues using graffiti in my own workflow.

**Comment by jbach** [comment]
I’m having the same issue, the first `POST` to `/graphql` is pending for quite some time, then fails. If I run the same query (`query IntrospectionQuery…`) via Postman, I get a proper response.

**Comment by neverfox** [comment]
I'd love to get this sorted since this is my first experience with considering graphQL for my next project (tired of REST endpoints proliferating like bunnies).

> Make sure that MongoDB is running.

Not only is it running, but I'm looking at the pre-populated data in MongoHub, so I know that was successful.

**Comment by jbach** [comment]
I have tracked down the problem: It works if I don’t use `bodyParser.json()` in my express app. Seems to be a problem with [graffiti](https://github.com/RisingStack/graffiti/blob/master/src/express/express.js#L2) / [co-body](https://github.com/cojs/co-body/blob/master/lib/json.js#L6) / [raw-body](https://github.com/stream-utils/raw-body). Can anyone confirm?

**Comment by neverfox** [comment]
Since the example uses Koa, that might be the implicit default body parser there.

**Comment by tothandras** [comment]
Can you try the examples in the [graffiti](https://github.com/RisingStack/graffiti/tree/master/example) project? I can't introduce the errors you are facing with.

**Comment by jbach** [comment]
@tothandras Regarding express, the problem arises if you [add JSON body parsing](https://gist.github.com/jbach/0ee869e63f187dc7ff54) as often suggested/needed by third party plugins. The workaround is to *not* use `body-parser` at root level or to exclude the `/graphql` endpoint via something like [express-unless](https://github.com/jfromaniello/express-unless). Would it be viable to check if the body still needs to be parsed [here](https://github.com/RisingStack/graffiti/blob/master/src/express/express.js#L23)?

**Comment by tothandras** [comment]
@jbach Would it be also viable to require body parsing by the plugin as other plugins do? :) Just drop it altogether and make it a requirement.

**Comment by jbach** [comment]
Probably. I checked up on [co-body](https://github.com/cojs/co-body), it doesn’t seem to prevent double parsing like [expressjs/body-parser](https://github.com/expressjs/body-parser/blob/master/lib/types/json.js#L92) [does](https://github.com/expressjs/body-parser/blob/master/lib/read.js#L44). I guess shifting that responsibility to the middleware would be *my* personal favorite.

**Comment by kaminskypavel** [comment]
+1 still having the same issue.

![graphql](https://cloud.githubusercontent.com/assets/4253088/12671702/26dd895e-c67a-11e5-976d-f15cd164f50e.png)

**Comment by KATT** [comment]
+1

**Comment by tothandras** [comment]
Thanks for the report! I will fix this tomorrow.

**Comment by tothandras** [comment]
@KATT @jbach @kaminskypavel @neverfox @dsifford 
Can you try version `3.0.1`. I've removed the body parsing from the middleware.

**Comment by kaminskypavel** [comment]
@tothandras , nope, unfortunately still getting `Uncaught TypeError: Cannot read property 'getQueryType' of undefined` .

also please update the readme to port 8081.

**Comment by jbach** [comment]
@tothandras, it does work for me. :+1:
@kaminskypavel, are you sure you updated both graffiti and graffiti-mongoose?

**Comment by kaminskypavel** [comment]
I did a clean checkout and install, am I missing something?

**Comment by KATT** [comment]
Eureka! I was struggling as well and noticed I was using npm `2.14.7` and node `4.2.1`.

Once I did a simple `nvm use 5` it kicked off for me. :tada:

**Comment by kaminskypavel** [comment]
what version of node was it?

**Comment by KATT** [comment]
```
$ node --version
v5.2.0
$ npm --version
3.3.12
```
